lib/assertions.js

```diff
@@ -1044,59 +1044,68 @@ module.exports = function (expect) {
                             if (subjectType.indent) {
                                 output.indentLines();
                             }
-                            changes.forEach(function (diffItem, index) {
+                            var packing = utils.packArrows(changes); // NOTE: Will have side effects in changes if the packing results in too many arrow lanes
+                            output.arrowsAlongsideChangeOutputs(packing, changes.map(function (diffItem, index) {
```

Comment:
It really has nothing to do with the PR, but I think we have crossed the line where we should split up the assertions in separates files (Not in this PR - but after). After we split the assertions into separate files, I think something like this would be much more readable if it was split into some more helper methods.